src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -525,29 +527,32 @@ private void SetAllExceptions(enum_EXCEPTION_STATE state)
         private void StepInternal(int threadId, enum_STEPKIND stepKind, enum_STEPUNIT stepUnit, string errorMessage)
         {
             // If we are already running ignore additional step requests
-            if (m_isStopped)
+            if (!m_isStopped)
+                return;
+
+
+            IDebugThread2 thread = null;
+            lock (m_threads)
             {
-                IDebugThread2 thread = null;
-                lock (m_threads)
+                if (!m_threads.TryGetValue(threadId, out thread))
                 {
-                    if (!m_threads.TryGetValue(threadId, out thread))
-                    {
-                        throw new AD7Exception(errorMessage);
-                    }
+                    throw new AD7Exception(errorMessage);
                 }
+            }
 
-                BeforeContinue();
-                ErrorBuilder builder = new ErrorBuilder(() => errorMessage);
-                m_isStepping = true;
-                try
-                {
-                    builder.CheckHR(m_program.Step(thread, stepKind, stepUnit));
-                }
-                catch (AD7Exception)
-                {
-                    m_isStopped = true;
-                    throw;
-                }
+            BeforeContinue();
+            ErrorBuilder builder = new ErrorBuilder(() => errorMessage);
+            m_isStepping = true;
+            if (!m_currentFrameHasSourceCode)
```

Comment:
>m_currentFrameHasSourceCode [](start = 17, length = 27)

1. We need to add a client check here -- we don't want to do this in full Visual Studio (or other IDEs that have full disassembly support)
2. We should probably check if the top frame of the specified thread has source, since the user can switch threads.

Reply:
This is used in VS?

Reply:
Yes.